tests/api.rs

```diff
@@ -0,0 +1,70 @@
+use assert_cmd::prelude::*;
+use predicates::prelude::*;
+use std::process::{Command, Stdio};
+
+use jsonrpsee::core::client::ClientT;
+use jsonrpsee::rpc_params;
+use std::time::Duration;
+use tokio::io::AsyncBufReadExt;
+
+#[test]
+fn must_provide_connection_string() -> Result<(), Box<dyn std::error::Error>> {
+    let mut cmd = Command::cargo_bin("tx-sitter")?;
+
+    cmd.arg("daemon");
+    cmd.assert()
+        .failure()
+        .stderr(predicate::str::contains(
+            "required arguments were not provided",
+        ))
+        .stderr(predicate::str::contains("CONNECTION_STRING"));
+
+    Ok(())
+}
+
+async fn read_until_match<R: tokio::io::AsyncRead + std::marker::Unpin>(read: R, target: &str) {
+    let mut bufread = tokio::io::BufReader::new(read);
+    loop {
+        let mut data = String::new();
+        bufread.read_line(&mut data).await.unwrap();
+        println!("data {data}");
+        if data.contains(target) {
+            break;
+        }
+    }
+}
+
+#[tokio::test]
+async fn starts_api() {
+    let cmd_path = assert_cmd::cargo::cargo_bin("tx-sitter");
+    let mut child = tokio::process::Command::new(cmd_path)
+        .arg(":memory:")
+        .arg("daemon")
+        .stderr(Stdio::piped())
+        .kill_on_drop(true) // caveat: tokio promises "best-effort" to reap this zombie
+        .spawn()
+        .expect("failed to spawn tx-sitter");
```

Comment:
Instead of spawning a new process, I think in Rust it's more common to construct an instance of `App` for testing. See for example here:

https://github.com/worldcoin/signup-sequencer/blob/a2dc5980b1ccc163e4bfc71c094353bd09847dcd/tests/integration-tests.rs#L303

This pattern works well in Rust since it promotes a pure-functional approach without global state.